### PR TITLE
Build ingress-nginx controller version 0.27.1 with FIPS support

### PR DESCRIPTION
**Pull request by alavrovinfb**
- re-built NGINX with FIPS support
- re-built ingress-nginx controller with FIPS support

### CODE REVIEW

**Review by Evgeniy-L — APPROVED** [review]
Looks ok to me

**Review by kd7lxl — COMMENTED** [review]
Do you have any suggestions on how this can be validated? Is there some build/test we should run with CI?

File: images/nginx/rootfs/build.sh [hunk 1]
```diff
@@ -600,8 +616,9 @@ writeDirs=( \
   /var/log/nginx \
 );
 
-addgroup -Sg 101 www-data
-adduser -S -D -H -u 101 -h /usr/local/nginx -s /sbin/nologin -G www-data -g www-data www-data
+groupadd -rg 101 www-data
+adduser -u 101 -M -d /usr/local/nginx \
```
Comment (daniel-garcia) [inline images/nginx/rootfs/build.sh:620]: are the userland tools different? is that why the args are different? if so, please comment upstream commands and write the updated commands separately

**Review by daniel-garcia — CHANGES_REQUESTED** [review]
The end of this (building) should be able to verify that openssl doesn't have md5 support and nginx lacks at least one crypto type that should have been removed.

**Comment by alavrovinfb** [comment]
> Do you have any suggestions on how this can be validated? Is there some build/test we should run with CI?

This image has been built based on Amazon Linux 2 that is conforms to FIPS and validated by NIST.
To test FIPS within CI the image must be deployed on the host system that configured with FIPS mode or we need to force FIPS mode enabling. Any way we can perform only some simple tests.

File: Makefile [hunk 2]
```diff
@@ -27,7 +28,7 @@ endif
 SHELL = /bin/bash
 
 # Use the 0.0 tag for testing, it shouldn't clobber any release builds
-TAG ?= master
+TAG ?= v0.27.1
```
[the remark this answers is not in the document]
Comment (alavrovinfb) [reply Makefile:31]: > eg circleci or jenkins
Do we use circleci or jenkins for infobloxopen repos? I think we use travis-ci for it.

[the remark this answers is not in the document]
Comment (kd7lxl) [reply Makefile:31]: In other infobloxopen repos, we have used Jenkins, Github Actions, and travis-ci.

[the remark this answers is not in the document]
Comment (alavrovinfb) [reply Makefile:31]: I see at least one project uses circle-ci https://app.circleci.com/pipelines/github/infobloxopen/seal

[the remark this answers is not in the document]
Comment (kd7lxl) [reply Makefile:31]: Sounds like there's no significant limitation on what build driver you can use for this.

**Comment by alavrovinfb** [comment]
Added simple OpenSSL fips mode test.
Some manual testing was done on FIPS enabled EC2 instance.

**Review by daniel-garcia — APPROVED** [review]

**Pull request merged**